dask_cloudprovider/providers/gcp/instances.py

```diff
@@ -0,0 +1,326 @@
+import asyncio
+import uuid
+
+import dask
+from dask_cloudprovider.providers.generic.vmcluster import (
+    VMCluster,
+    VMInterface,
+    SchedulerMixin,
+    WorkerMixin,
+)
+
+from dask_cloudprovider.utils.socket import is_socket_open
+
+try:
+    import googleapiclient.discovery
+    from googleapiclient.errors import HttpError
+    from google.auth.exceptions import DefaultCredentialsError
+except ImportError as e:
+    msg = (
+        "Dask Cloud Provider GCP requirements are not installed.\n\n"
+        "Please either conda or pip install as follows:\n\n"
+        "  conda install -c conda-forge google-api-python-client  # either conda install\n"
+        '  python -m pip install "dask-cloudprovider[gcp]" --upgrade  # or python -m pip install'
+    )
+    raise ImportError(msg) from e
+
+
+class GCPInstance(VMInterface):
+    def __init__(
+        self,
+        cluster,
+        name,
+        config=None,
+        zone=None,
+        projectid=None,
+        machine_type=None,
+        filesystem_size=None,
+        source_image=None,
+        docker_image=None,
+        ngpus=None,
+        gpu_type=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.cluster = cluster
+        self.name = name
+        self.config = config
+        self.projectid = projectid or self.config.get("projectid")
+        self.zone = zone or self.config.get("zone")
+
+        self.machine_type = machine_type or self.config.get("machine_type")
+
+        self.source_image = source_image or self.config.get("source_image")
+        self.docker_image = docker_image or self.config.get("docker_image")
+        self.filesystem_size = filesystem_size or self.config.get("filesystem_size")
+        self.ngpus = ngpus or self.config.get("ngpus")
+        self.gpu_type = gpu_type or self.config.get("gpu_type")
+
+        self.general_zone = "-".join(self.zone.split("-")[:2])  # us-east1-c -> us-east1
```

Comment:
We should add a default zone otherwise this fails with default config.

```python-traceback
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-16-03d091edaadb> in async-def-wrapper()

/opt/conda-environments/rapids-stable/lib/python3.7/site-packages/distributed/deploy/spec.py in _()
    382         async def _():
    383             if self.status == Status.created:
--> 384                 await self._start()
    385             await self.scheduler
    386             await self._correct_state()

~/Projects/dask/dask-cloudprovider/dask_cloudprovider/providers/generic/vmcluster.py in _start(self)
    115             "Hang tight! ",
    116         ):
--> 117             await super()._start()
    118 
    119     def render_cloud_init(self, *args, **kwargs):

/opt/conda-environments/rapids-stable/lib/python3.7/site-packages/distributed/deploy/spec.py in _start(self)
    299         if isinstance(cls, str):
    300             cls = import_term(cls)
--> 301         self.scheduler = cls(**self.scheduler_spec.get("options", {}))
    302 
    303         self.status = Status.starting

~/Projects/dask/dask-cloudprovider/dask_cloudprovider/providers/gcp/instances.py in __init__(self, *args, **kwargs)
    219 
    220     def __init__(self, *args, **kwargs):
--> 221         super().__init__(*args, **kwargs)
    222         self.command = "dask-scheduler --host 0.0.0.0"
    223         self.name = f"dask-scheduler-{str(uuid.uuid4())[:8]}"

~/Projects/dask/dask-cloudprovider/dask_cloudprovider/providers/gcp/instances.py in __init__(self, cluster, name, config, zone, projectid, machine_type, filesystem_size, source_image, docker_image, ngpus, gpu_type, **kwargs)
     57         self.gpu_type = gpu_type or self.config.get("gpu_type")
     58 
---> 59         self.general_zone = "-".join(self.zone.split("-")[:2])  # us-east1-c -> us-east1
     60 
     61     def create_gcp_config(self):

AttributeError: 'NoneType' object has no attribute 'split'
```